operator/controllers/csiscaleoperator_controller.go

```diff
@@ -164,12 +164,12 @@ func (r *CSIScaleOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		logger.V(1).Info("Updated resource status.", "Status", cr.Status)
 	}()
 
-	r.Scheme.Default(instanceUnwrap)
-	err = r.Client.Update(context.TODO(), instanceUnwrap)
-	if err != nil {
-		logger.Error(err, "Reconciler Client.Update() failed")
-		return ctrl.Result{}, err
-	}
+	// r.Scheme.Default(instanceUnwrap)
+	// err = r.Client.Update(context.TODO(), instanceUnwrap)
```

Comment:
With recent changes in CSI dev branch,
Default values are removed from all the fields except `nodeSelectors` and `secureSslMode`.
Also `omitempty` it added in all the spec fields to ensure unused fields do not get encoded with default values like ""(empty string).

I agree the commented code in the PR, was the updating the CR with default values in certain fields resulting in increase in `metadata.generation`
However with above mentioned changes in latest CSI code, this commented code should not be setting default values anymore.
I can confirm this, because I don't see `metadata.generation` increasing anymore because of it. 

I feel commenting the update logic here is fine, this piece of code is not really required however,
I could be missing some thing here, but @tychristensen  I am eager to know what exactly is fixed by commenting this code that helps fix the linked CNSA issue? 
As I mentioned default values like empty string are no longer set by the commented code.

Reply:
The CR is updated at two more places, 
While adding finalizer and while adding CGPrefix in CR.
`metadata.generation` is 2 at starting which means spec was updated once, this could be because of CGPrefix spec change.